packaging/systemd/cri-docker.service

```diff
@@ -8,7 +8,7 @@ Requires=cri-docker.socket
 
 [Service]
 Type=notify
-ExecStart=/usr/bin/cri-dockerd --v=10 --networkplugin="" --logtostderr
+ExecStart=/usr/bin/cri-dockerd --container-runtime-endpoint fd:// --v=10 --networkplugin="" --logtostderr
```

Comment:
In my setup this fails (built using this branch).

```
Sep 21 15:36:25 node01 cri-dockerd[788]: F0921 15:36:25.371476     788 server.go:58] unknown flag: --networkplugin
Sep 21 15:36:25 node01 systemd[1]: cri-docker.service: main process exited, code=exited, status=255/n/a
```

```
Sep 21 15:36:25 node01 cri-dockerd[788]: Usage:
Sep 21 15:36:25 node01 cri-dockerd[788]: cri-dockerd [flags]
Sep 21 15:36:25 node01 cri-dockerd[788]: Flags:
...
Sep 21 15:36:25 node01 cri-dockerd[788]: --network-plugin string                   <Warning: Alpha feature> The name of the network plugin to be invoked for various events in kubelet/pod lifecycle.
Sep 21 15:36:25 node01 cri-dockerd[788]: --network-plugin-mtu int32                <Warning: Alpha feature> The MTU to be passed to the network plugin, to override the default. Set to 0 to use the default 1460 MTU.
...
```

Reply:
See #9

Reply:
Also note that on centos7 at least the `""` fails for me. If I leave it as `--network-plugin=` it works better.

Reply:
systemd-escape bug, perhaps ?

Reply:
Perhaps. it resulted with this in journalctl logs..

```
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341034   32024 flags.go:59] FLAG: --alsologtostderr="false"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341072   32024 flags.go:59] FLAG: --cni-bin-dir=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341076   32024 flags.go:59] FLAG: --cni-cache-dir=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341079   32024 flags.go:59] FLAG: --cni-conf-dir=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341081   32024 flags.go:59] FLAG: --container-runtime-endpoint="fd://"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341083   32024 flags.go:59] FLAG: --docker-endpoint="unix:///var/run/docker.sock"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341086   32024 flags.go:59] FLAG: --dockershim-root-directory="/var/lib/dockershim"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341088   32024 flags.go:59] FLAG: --help="false"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341091   32024 flags.go:59] FLAG: --image-pull-progress-deadline="1m0s"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341094   32024 flags.go:59] FLAG: --log-backtrace-at=":0"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341098   32024 flags.go:59] FLAG: --log-dir=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341100   32024 flags.go:59] FLAG: --log-file=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341102   32024 flags.go:59] FLAG: --log-flush-frequency="5s"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341105   32024 flags.go:59] FLAG: --logtostderr="true"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341108   32024 flags.go:59] FLAG: --network-plugin="\"\""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341112   32024 flags.go:59] FLAG: --network-plugin-mtu="0"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341116   32024 flags.go:59] FLAG: --pod-cidr=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341119   32024 flags.go:59] FLAG: --pod-infra-container-image="k8s.gcr.io/pause:3.1"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341121   32024 flags.go:59] FLAG: --runtime-cgroups=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341124   32024 flags.go:59] FLAG: --stderrthreshold="2"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341126   32024 flags.go:59] FLAG: --v="10"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341128   32024 flags.go:59] FLAG: --version="false"
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341130   32024 flags.go:59] FLAG: --vmodule=""
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341134   32024 client.go:77] Connecting to docker on unix:///var/run/docker.sock
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.341146   32024 client.go:94] Start docker client with request timeout=0s
Sep 21 15:41:18 node01 cri-dockerd[32024]: I0921 15:41:18.342009   32024 docker_service.go:242] "Hairpin mode is set" hairpinMode=none
Sep 21 15:41:18 node01 cri-dockerd[32024]: W0921 15:41:18.342058   32024 cni.go:239] Unable to update cni config: no networks found in
Sep 21 15:41:18 node01 cri-dockerd[32024]: W0921 15:41:18.343966   32024 hostport_manager.go:72] The binary conntrack is not installed, this can cause failures in network connection cleanup.
Sep 21 15:41:18 node01 cri-dockerd[32024]: W0921 15:41:18.343976   32024 hostport_manager.go:72] The binary conntrack is not installed, this can cause failures in network connection cleanup.
Sep 21 15:41:18 node01 cri-dockerd[32024]: F0921 15:41:18.344001   32024 server.go:88] didn't find compatible CNI plugin with given settings &{HairpinMode:none NonMasqueradeCIDR:10.0.0.0/8 PluginName:"" PluginBinDirString: PluginBinDirs:[] PluginConfDir: PluginCacheDir: MTU:0}: network plugin "\"\"" not found
```

Reply:
ah, maybe same end effect (but with messier logs) then. good catch!

Reply:
Also note that on CentOS7 the unit stalls indefinitely with issuing a start/restart. Are sure `Type=notify` is correct? I have confirmed the service is up and running using `critctl` so it's 'just' systemd isn't believing so.

Reply:
To be clear about the double quotes, when I have this in the unit file: `ExecStart=/usr/bin/cri-dockerd --container-runtime-endpoint fd:// --v=10 --network-plugin= --logtostderr` the logs show this `I0921 15:57:53.007953    8996 flags.go:59] FLAG: --network-plugin=""`

Reply:
Right, it says `FLAG: --network-plugin=""` on a modern systemd (_without_ changing the unit, that is)

`ExecStart=/usr/bin/cri-dockerd --container-runtime-endpoint fd:// --network-plugin=""`

Reply:
The hang is the same, though. All the other container runtimes use `notify`, so maybe there is some more code missing ?

Reply:
I have no idea..happy to try ideas if anyone has some to share. I'm not sure what's supposed to send the notify and/or how etc.

Reply:
Will add another PR for adding sd_notify, using the same module: github.com/coreos/go-systemd/v22

Reply:
OK awesome!

Reply:
See #13